docs(dart): Add feature flags documentation (#13391)
getsentry/sentry-docs: includes/dart-integrations/firebase-remote-config.mdx

---
title: Firebase Remote Config
description: "Learn more about the Sentry Firebase Remote Config integration for the Dart SDK."
sidebar_order: 6
platforms:
  - flutter
---

The `sentry_firebase_remote_config` integration provides [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) support for Sentry thus providing insight into feature flag evaluations.

## Behavior

- Adding the `SentryFirebaseRemoteConfig` integration will automatically track feature flag evaluations.
- Only boolean values are supported.

## Prerequisites

> 1. [Sentry SDK](/platforms/dart/#configure) version `9.0.0` or higher is installed.

Reviewer: ```suggestion
1. [Sentry SDK](/platforms/dart/#configure) version `9.0.0`.
```

2. Firebase Remote Config is set up.

## Install

To use the `SentryFirebaseRemoteConfig` integration, add the `sentry_firebase_remote_config` dependency.

```yml {filename:pubspec.yaml}
dependencies:
  sentry: ^9.0.0
  sentry_firebase_remote_config: ^9.0.0
```

## Configure

Add the `SentryFirebaseRemoteConfig` integration to the Sentry SDK initialization. The integration will automatically call `await remoteConfig.activate();` when the config is updated.
If you don't want this behavior, you can set `activateOnConfigUpdated` to `false`.

```dart
await SentryFlutter.init(
  (options) {
    options.addIntegration(
      SentryFirebaseRemoteConfigIntegration(
        firebaseRemoteConfig: firebaseRemoteConfig,
        activateOnConfigUpdated: false,
      ),
    );
  },
);

> ```

Reviewer 2: > If you don't want this behavior, you can set `activateOnConfigUpdated` to `false`.

Can we split this up into another section? 

The first is `Configure` and then the subsection would be something like `Remote Config Update` and then explain the `activateOnConfigUpdated` there.

It's confusing to me that `activateOnConfigUpdated: false` is being set here

## Verify & View

> ### 1. Update Firebase Remote Config value

Reviewer: ```suggestion
### 1. Update Firebase Remote Config Value
```

> Update a boolean [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) value to true. Keep in mind that values will be evaluated as bool if there string values are `true`, `false`, `1`, or `0`. So keep in mind that numerical values of `1` or `0` will be evaluated as `true` or `false` boolean values.

Reviewer 3: ```suggestion
Update a boolean [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) value to true. Keep in mind that values will be evaluated as bool if their string values are `true`, `false`, `1`, or `0`. So numerical values of `1` or `0` will be evaluated as `true` or `false` boolean values respectively.
```

### 2. View the Recorded Feature Flag Evaluation on Sentry.io

To view the recorded feature flag evaluation, log into [sentry.io](https://sentry.io) and open your project.

> The recorded feature flag evaluations will be visible in the event context.

Reviewer: I think a screenshot here would be helpful.

---
title: Firebase Remote Config
description: "Learn more about the Sentry Firebase Remote Config integration for the Dart SDK."
sidebar_order: 6
platforms:
  - flutter
---

The `sentry_firebase_remote_config` integration provides [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) support for Sentry thus providing insight into feature flag evaluations.

## Behavior

> - Adding the `SentryFirebaseRemoteConfig` integration will automatically track feature flag evaluations for boolean Firebase Remote Config values.
> - Only boolean values are supported.

Reviewer 2: ```suggestion
- Adding the `SentryFirebaseRemoteConfig` integration will automatically track feature flag evaluations.
- Only boolean values are supported.
```

> - The integration is available in <PlatformLink to="/">Sentry SDK</PlatformLink> version 9.0.0 or higher.

Reviewer 2: Let's mention that only booleans are currently supported

> - The integration is available in <PlatformLink to="/">Sentry SDK</PlatformLink> version 9.0.0 or higher.

Reviewer 2: ```suggestion
```

We can remove this and add it to the prerequisite (see comment below)

## Prerequisites

> 1. The Sentry [Dart](/platforms/dart/#configure) SDK is initialized.

Reviewer 2: ```suggestion
1. SDK version `9.0.0` or higher is installed.

```

Reviewer 2: I know we mention the initialized a couple times in other integrations. I'm actually thinking of removing them as imo they dont bring a lot of value

2. Firebase Remote Config is set up.

## Install

To use the `SentryFirebaseRemoteConfig` integration, add the `sentry_firebase_remote_config` dependency.

```yml {filename:pubspec.yaml}
dependencies:
  sentry: ^9.0.0
  sentry_firebase_remote_config: ^9.0.0
```

## Configure

Add the `SentryFirebaseRemoteConfig` integration to the Sentry SDK initialization.

```dart
import 'package:sentry_flutter/sentry_flutter.dart';
import 'package:sentry_firebase_remote_config/sentry_firebase_remote_config.dart';

... // Initialize Firebase Remote Config

await SentryFlutter.init(
  (options) {
    options.dsn = 'https://example@sentry.io/example';
    options.addIntegration(
      SentryFirebaseRemoteConfigIntegration(
        firebaseRemoteConfig: firebaseRemoteConfig,

>         // Don't call `await remoteConfig.activate();` when firebase config is updated. Per default this is true.

Reviewer 4: You're telling me not to call `await remoteConfig.activate()` depending on the value of the config below?
if so, I shouldn't call when the value is set to `false`? But the default is `true`?

Might be just me but this is a bit confusing

Reviewer 2: I believe what it means is that they don't have to call `await remoteConfig.activate()` manually since the SDK does it already and that you can override this behaviour if you set `activateOnConfigUpdated: false`

Yea sounds confusnig, let's update this, maybe we can explain this in another section below

        activateOnConfigUpdated: false,
      ),
    );
  },
);

> ```

Reviewer 2: imo we can remove the comment, the import and the options.dsn

## Verify & View

### 1. Update Firebase Remote Config value

Update a boolean [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) value to true. Keep in mind that values will be evaluated as bool if there string values are `true`, `false`, `1`, or `0`. So keep in mind that numerical values of `1` or `0` will be evaluated as `true` or `false` boolean values.

### 2. View the Recorded Feature Flag Evaluation on Sentry.io

To view the recorded feature flag evaluation, log into [sentry.io](https://sentry.io) and open your project.

> The recorded feature flag evaluations will be visible in the event context.

Reviewer 4: One note for the future: We don't support change tracking for Firebase **yet**.
But once we do, we'll add it to here: https://docs.sentry.io/product/issues/issue-details/feature-flags/#change-tracking

Reviewer 5: Bruno can you clarify - is the wording in the PR ok? because while change tracking is not there, this paragraph talks about [evaluation tracking](https://docs.sentry.io/product/issues/issue-details/feature-flags/#evaluation-tracking)

Reviewer 4: It was just a note for the future. But that makes sense, thanks for pointing out.

It turns out it's not super easy to add Firebase support either and might require the customer to add some firebase lambda in order to call our tracking endpoint. @michellewzhang wrote about it here: https://www.notion.so/sentry/firebase-feature-flag-webhook-1e38b10e4b5d805fb057d2e990ea45e8

I wonder if this is something @aliu39 might be keen to tackle in the future though.

---
title: Firebase Remote Config
description: "Learn more about the Sentry Firebase Remote Config integration for the Dart SDK."
sidebar_order: 6
platforms:
  - flutter
---

The `sentry_firebase_remote_config` integration provides [Firebase Remote Config](https://firebase.google.com/docs/remote-config/get-started?platform=flutter) support for Sentry, providing insight into feature flag evaluations.

## Behavior

- Adding the `SentryFirebaseRemoteConfig` integration will automatically track feature flag evaluations.
- Only boolean values are supported.

## Prerequisites

> 1. [Sentry SDK](/platforms/dart/#configure) version `9.0.0`.

Reviewer 2: ```suggestion
1. [Sentry SDK](/platforms/dart/#configure) version `9.0.0` or higher.
```
